src/Pact/Eval.hs

```diff
@@ -217,8 +217,8 @@ loadModule m@Module{..} bod1 mi g0 = do
         second HM.fromList <$> foldM doDef (g0,[]) bd
       t -> evalError (_tInfo t) "Malformed module"
   evaluatedDefs <- evaluateDefs mi modDefs1
-  evaluateConstraints mi m evaluatedDefs
-  let md = ModuleData m evaluatedDefs
+  (m', solvedDefs) <- evaluateConstraints mi m evaluatedDefs
```

Comment:
we have to pass along the module to modify the body `meta` declarations

Reply:
On a meta-level (no pun intended), it's surprising / disappointing to me that we need to do any bookkeeping during evaluation for this to work.

Reply:
Can we just erase models at runtime?

Reply:
Well, I think there's always going to be _some_ bookkeping, unfortunately. I'm happy to discuss future changes we can make to get around having to do so much 😄

Reply:
I would like to discuss how we can reduce the bookkeeping for features that have no runtime effect